internal/cmd/commands/authtokenscmd/funcs.go

```diff
@@ -1,13 +1,60 @@
 package authtokenscmd
 
 import (
+	"errors"
 	"fmt"
+	"strings"
 	"time"
 
 	"github.com/hashicorp/boundary/api/authtokens"
 	"github.com/hashicorp/boundary/internal/cmd/base"
+	"github.com/hashicorp/boundary/sdk/strutil"
 )
 
+func init() {
+	extraFlagsHandlingFunc = extraFlagsHandlingFuncImpl
+}
+
+func extraFlagsHandlingFuncImpl(c *Command, _ *base.FlagSets, _ *[]authtokens.Option) bool {
+	if c.Func != "delete" && c.Func != "read" {
+		if strutil.StrListContains(flagsMap[c.Func], "id") && c.FlagId == "" {
+			c.PrintCliError(errors.New("ID is required but not passed in via -id"))
+			return false
+		}
+		return true
+	}
+
+	if c.FlagId == "" {
+		fmt.Printf("No ID provided; %s the stored token? (Pass an ID of %q to suppress this question.) y/n: ", c.Func, "self")
+		var yesNo string
+		fmt.Scanf("%s", &yesNo)
+		switch yesNo {
+		case "y", "Y":
+			c.FlagId = "self"
```

Comment:
"self" is used a few times, should it be a const?

Reply:
Changed